liferay-forms/liferay-portal
# LPS-200278 Draft DDMFormInstanceVersion and DDMStructureVersion cannot be approved

Hi team, can you please review these changes? Thank you!

### Motivation
https://liferay.atlassian.net/browse/LPS-200278

In publications, we don't allow publishing a publication with draft changes. While doing testing, we noticed that with every form update, it creates a new DDMFormInstanceVersion and DDMStructureVersion. When a form rule is added, the form is saved as draft with the version objects' status set to `draft`. However, when the form is published, the draft versions are not updated, and new ones are created instead.

### Proposed Solution
Version control is handled differently in forms. Usually, we only create a new version if the current version is published/approved. We keep updating the draft until it's published/approved. Can we apply the same behavior to DDMFormInstanceVersion and DDMStructureVersion so that we don't just keep draft versions in the db without being able to update them? 

### Steps to verify
1. Add a new form and add a rule. Don't save or publish the form yet
2. Verify in the database that DDMFormInstanceVersion and DDMStructureVersion are created with `draft` status = 2
3. Publish the form
4. The draft DDMFormInstanceVersion and DDMStructureVersion are still in the database, and new ones with approved status = 1 are created.
